### DISCUSSION TITLE
Kustomization gets "failed to download artifact" error on CCE cluster

### DISCUSSION THREAD

**Discussion by wangyuan249** [fluxcd/flux2#3772] [Q&A]
Hi everyone~,  I'm trying to install flux on huawei CCE cluster and deploy kustomization configuration,
Due to the permission control reasons, I change the source-controller  `STORAGE_PATH`  from  `"/data"`  to `"/var/paas/flux"`
Like this:
![image](https://user-images.githubusercontent.com/35907099/230614754-f86709ff-715b-4355-898f-c7064e68bc94.png)
And the source-controller runs normally and gitrepo can pull the artifact down in container. 

```
kubectl get gitrepo
NAME            URL                                       AGE    READY   STATUS
lwj-repo01      https://github.com/GitHubxsy/podinfo      10h    True    stored artifact for revision 'master@sha1:13f5b1b40600554990b7dfbcfd31da27f1e727e3'
podinfo         https://github.com/stefanprodan/podinfo   5m6s   True    stored artifact for revision 'master@sha1:58726f0bd2755f279d2aed9bbfa0adc0b84cb3bc'
```
Howerver, I get this error from the kustomization-controller log : 
```
{"level":"error","ts":"2023-04-06T20:23:34.313+0800","msg":"Reconciliation failed after 3m35.050486112s, next try in 1m0s",
"controller":"kustomization","controllerGroup":"kustomize.toolkit.fluxcd.io",
"controllerKind":"Kustomization","Kustomization":{"name":"lwj-configset01","namespace":"default"},
"namespace":"default","name":"lwj-configset01",
"reconcileID":"ac17725a-c2bd-4555-b203-9e2dfc840dde",
"revision":"master@sha1:13f5b1b40600554990b7dfbcfd31da27f1e727e3",
"error":"failed to download archive, 
error: GET http://localhost:9090/gitrepository/default/lwj-repo01/13f5b1b40600554990b7dfbcfd31da27f1e727e3.tar.gz 
giving up after 10 attempt(s): 
Get \"http://localhost:9090/gitrepository/default/lwj-repo01/13f5b1b40600554990b7dfbcfd31da27f1e727e3.tar.gz\": 
dial tcp [::1]:9090: connect: connection refused"}
```

```
kubectl get ks
NAME               AGE     READY     STATUS
lwj-configset01    3d5h    Unknown   Reconciliation in progress
podinfo            4m34s   False     failed to download archive, error: GET http://localhost:9090/gitrepository/default/podinfo/58726f0bd2755f279d2aed9bbfa0adc0b84cb3bc.tar.gz giving up after 10 attempt(s): 
Get "http://localhost:9090/gitrepository/default/podinfo/58726f0bd2755f279d2aed9bbfa0adc0b84cb3bc.tar.gz": dial tcp [::1]:9090: connect: connection refused
```

my flux version is :

```
flux version
flux: v0.40.0
helm-controller: v0.30.0
kustomize-controller: v0.34.0
notification-controller: v0.32.0
source-controller: v0.35.1
```

May I know why the error occurred or something configed wrong?
Looking forward to any suggestion, thanks!

**Comment by makkes** [comment]
This doesn't look right:
```
GET http://localhost:9090/gitrepository` 
```
kustomize-controller shouldn't fetch from `localhost:9090` but from the service name `source-controller.flux-system.svc.cluster.local.`. 

Please paste the output of
```
kubectl get gitrepo lwj-repo01 -o jsonpath={.status.artifact.url}
```

**Reply by wangyuan249** [reply]
Thank you for your reply. I have also noticed that the service address which kustomize-controller  —> source-controller is not configured correctly.

And the wrong configuration is caused  by I want to find an official Helmchart to install Flux2, but the chart in this repo `https://github.com/fluxcd-community/helm-charts`  is not quite right.

**Reply by wangyuan249** [reply]
Specifically the yaml in
```
https://github.com/fluxcd-community/helm-charts/blob/main/charts/flux2/templates/kustomize-controller.yaml
```
lack of 
```
- name: SOURCE_CONTROLLER_LOCALHOST
          value: source-controller.flux-system.svc.cluster.local.:9090
```

and the yaml  in 

```
https://github.com/fluxcd-community/helm-charts/blob/main/charts/flux2/templates/source-controller-service.yaml
```
should be configured like
```
spec:
  ports:
  - name: http
    port: 9090
    protocol: TCP
    targetPort: 9090
```
rather than 
![image](https://github.com/fluxcd/flux2/assets/35907099/f2b0d78f-71d1-4dfe-917e-0f191de42d9d)

**Comment by wangyuan249** [comment]
Thank you very much for your guidance！Our team has decided to integrate Flux into our distributed cloud platform UCS(Which is benchmarking against Google Anthos and Microsoft Arc.) for multi-cluster and multi-cloud deployment. 
And I hope there would be more exchange opportunities about gitops on multi-cluster/fleet in the future ~
